packages/user-facing-text/api.d.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Build a message from given template and arguments.
+ */
+export function buildMessage(template: string, ...params: string[]): string;
+
+type validationText = {
+  msg: string;
+  relatedInfo?: string;
```

Comment:
maybe add to the Validations type:
```json
{
  NON_UNIQUE_ID_RELATED_INFO : { 
                             msg: "identical ID also used here",
                             code: NaN // Not A Number
                          }
}
```

instead of adding it to the validations text?

Reasoning:
- Considering it is optional, the usage is not straightforward (e.g `??`).
- If more hierarchy does not seem to help with the modeling, than we can make it more "flat. It would be a little ugly on a specific property (NaN), but more maybe cleaner on an "regular" property.